daemon/graphdriver/counter.go

```diff
@@ -1,32 +1,69 @@
 package graphdriver
 
-import "sync"
+import (
+	"sync"
+
+	"github.com/docker/docker/pkg/mount"
+)
+
+type minfo struct {
+	check bool
+	count int
+}
 
 // RefCounter is a generic counter for use by graphdriver Get/Put calls
 type RefCounter struct {
-	counts map[string]int
+	counts map[string]*minfo
 	mu     sync.Mutex
 }
 
 // NewRefCounter returns a new RefCounter
 func NewRefCounter() *RefCounter {
-	return &RefCounter{counts: make(map[string]int)}
+	return &RefCounter{counts: make(map[string]*minfo)}
 }
 
 // Increment increaes the ref count for the given id and returns the current count
-func (c *RefCounter) Increment(id string) int {
+func (c *RefCounter) Increment(path string) int {
 	c.mu.Lock()
-	c.counts[id]++
-	count := c.counts[id]
+	m := c.counts[path]
+	if m == nil {
+		m = &minfo{check: true}
```

Comment:
Should the initial state of check be "false" instead of "true"? IIUC, if a path is not 
in the c.counts map (m == nil), you want to check if it is mounted or not. Have I misunderstood it.